zellij-server/src/ui/overlay/mod.rs

```diff
@@ -28,9 +29,11 @@ pub enum OverlayType {
 }
 
 impl Overlayable for OverlayType {
-    fn generate_overlay(&self, size: Size) -> String {
+    fn generate_overlay(&self, size: Size) -> Result<String> {
         match &self {
-            OverlayType::Prompt(prompt) => prompt.generate_overlay(size),
+            OverlayType::Prompt(prompt) => prompt
+                .generate_overlay(size)
+                .context("failed to generate overlay for overlay type"),
```

Comment:
Since we're implementing this trait below an a real type called `Overlay` we should word the context a little different here. Given the docstring for `Overlayable`, what do you think about the following?

```suggestion
                .context("failed to generate VTE output from overlay type"),
```